src/main/scala/chisel3/tester/ChiselUtestTester.scala

```diff
@@ -0,0 +1,74 @@
+package chisel3.tester
+
+import utest.TestSuite
+import chisel3.MultiIOModule
+import chisel3.tester.internal._
+import chisel3.tester.experimental.sanitizeFileName
+
+import firrtl.AnnotationSeq
+
+/**
+ * Using utest as test framework
+ * {{{
+ *   // define test spec in trait
```

Comment:
I would suggest putting this code into a unit test

Reply:
It's a little tricky, since we wanna use scalatest to test utest framework.

Reply:
Seems not too tricky, here is the result:
```
[info] ScalaTest
[info] Run completed in 40 seconds, 304 milliseconds.
[info] Total number of tests run: 88
[info] Suites: completed 26, aborted 0
[info] Tests: succeeded 88, failed 0, canceled 0, ignored 1, pending 0
[info] All tests passed.
[info] utest
[info] Tests: 1, Passed: 1, Failed: 0
```